[regression] 'GroupAction' object has no attribute 'actions'

<!--
For general questions, please ask on ROS answers: https://answers.ros.org, make sure to include at least the `ros2` tag and the rosdistro version you are running, e.g. `ardent`.
For general design discussions, please post on discourse: https://discourse.ros.org/c/ng-ros
Not sure if this is the right repository? Open an issue on https://github.com/ros2/ros2/issues
For Bug report or feature requests, please fill out the relevant category below
-->

## Bug report

**Required Info:**

- Operating System: Ubuntu 18.04
- Installation type: source
- Version or commit hash: 4eb26874260bab821863d28e52fbcecd25553f07
- DDS implementation: Fast-RTPS

#### Steps to reproduce issue
<!-- Detailed instructions on how to reliably reproduce this issue http://sscce.org/
``` code that can be copy-pasted is preferred ``` -->

Launch the following script:

```python

from launch import LaunchDescription
from launch.actions import GroupAction
from launch.actions import LogInfo

import launch.actions
import launch_ros.actions


def generate_launch_description():
    return LaunchDescription([
        GroupAction([
            launch.actions.LogInfo(msg=['Doing Action 1'])
        ])
    ])

```

#### Expected behavior
Terminal output with message `Doing Action 1`.

#### Actual behavior
Terminal output is:

``` bash
[ERROR] [launch]: Caught exception in launch (see debug for traceback): 'GroupAction' object has no attribute 'actions'
```

Replacing `LogInfo` with a plain `Action` results on the same error.

--- a/launch/launch/actions/group_action.py
+++ b/launch/launch/actions/group_action.py
@@ -71,7 +71,7 @@ class GroupAction(Action):
 
     def get_sub_entities(self) -> List[LaunchDescriptionEntity]:
         """Return subentities."""
-        return self.actions
+        return self.__actions
 
     def execute(self, context: LaunchContext) -> Optional[List[LaunchDescriptionEntity]]:
         """Execute the action."""

--- a/launch/launch/actions/group_action.py
+++ b/launch/launch/actions/group_action.py
@@ -58,6 +58,7 @@ class GroupAction(Action):
             self.__launch_configurations = launch_configurations
         else:
             self.__launch_configurations = {}
+        self.__actions_to_return: Optional[List] = None
 
     @classmethod
     def parse(cls, entity: Entity, parser: Parser):
@@ -71,13 +72,20 @@ class GroupAction(Action):
 
     def get_sub_entities(self) -> List[LaunchDescriptionEntity]:
         """Return subentities."""
-        return self.__actions
+        if self.__actions_to_return is None:
+            self.__actions_to_return = []  # type: List[Action]
+            self.__actions_to_return += [
+                SetLaunchConfiguration(k, v) for k, v in self.__launch_configurations.items()
+            ]
+            self.__actions_to_return += list(self.__actions)
+            if self.__scoped:
+                self.__actions_to_return = [
+                    PushLaunchConfigurations(),
+                    *self.__actions_to_return,
+                    PopLaunchConfigurations()
+                ]
+        return self.__actions_to_return
 
     def execute(self, context: LaunchContext) -> Optional[List[LaunchDescriptionEntity]]:
         """Execute the action."""
-        actions = []  # type: List[Action]
-        actions += [SetLaunchConfiguration(k, v) for k, v in self.__launch_configurations.items()]
-        actions += list(self.__actions)
-        if self.__scoped:
-            return [PushLaunchConfigurations(), *actions, PopLaunchConfigurations()]
-        return actions
+        return self.get_sub_entities()

